distributed/tests/test_scheduler.py

```diff
@@ -370,25 +370,38 @@ async def test_clear_events_client_removal(c, s, a, b):
         assert time() < start + 2
 
 
-@gen_cluster()
-async def test_add_worker(s, a, b):
-    w = Worker(s.address, nthreads=3)
-    w.data["x-5"] = 6
-    w.data["y"] = 1
```

Comment:
This is the test that probably should have caught the regression, but it didn't because `Scheduler.add_worker` only [cares about `nbytes`](https://github.com/gjoseph92/distributed/blob/e23f0eaeef8029cc69dfda614da6fa0d194565a2/distributed/scheduler.py#L3655). By just adding data, but not corresponding TaskStates, `nbytes` was empty in the worker's reconnection message: https://github.com/dask/distributed/blob/e23f0eaeef8029cc69dfda614da6fa0d194565a2/distributed/worker.py#L1110-L1117

The [`keys` field](https://github.com/dask/distributed/blob/e23f0eaeef8029cc69dfda614da6fa0d194565a2/distributed/worker.py#L1107) of this message is unused by the scheduler and should probably be removed. It should be redundant to `nbytes` anyway.